cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,53 @@
-/// <reference types='cypress' />
+const { generateUser } = require('./support/commands');
 
 describe('Student Registration page', () => {
   before(() => {
-
+    cy.visit('https://demoqa.com/automation-practice-form');
   });
 
-  it('', () => {
+  it('Enter valid data', () => {
+    const {
+      email, username, phone, lastname,
+      address, gender, hobby, subject
+    } = generateUser();
+
+    cy.get('#firstName').type(username);
+    cy.get('#lastName').type(lastname);
+    cy.get('#userEmail').type(email);
+    cy.get('#userNumber').type(phone);
+    cy.get('#currentAddress').type(address);
+
+    // Update gender and hobby selections
+    cy.get(`#gender-wrapper input[value="${gender}"]`).check(); // Assuming radio buttons
+    cy.get(`#hobbies-wrapper input[value="${hobby}"]`).check(); // Assuming checkboxes
```

Comment:
The selectors used here are quite brittle and could easily break if the structure of the HTML changes. Consider adding data-test attributes to your HTML elements to make your tests more resilient to changes in the structure or styling of your application.